### PR TITLE
Update Metalava

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

This PR updates Metalava

Closes https://github.com/mapbox/mapbox-navigation-android/issues/5852

### CODE REVIEW

File: scripts/update_metalava.sh [hunk 1]
```diff
@@ -16,10 +16,19 @@ fi
     echo " Done"
 
     echo -n "Building…"
+    # Dev branch has JAVA_HOME defined in the project, if building from development branches,
+    # uncomment below to exclude JAVA_HOME export and use JAVA_HOME from local machine.
+#    sed '/^export JAVA_HOME=/d' gradlew > gradlew.temp
+#    mv gradlew.temp gradlew
+#    chmod a+x gradlew
```
Comment (Guardiola31337) [inline scripts/update_metalava.sh:23]: `$> make update-metalava` was failing because of ^^ cc @RingerJK @LukasPaczos

File: scripts/update_metalava.sh [hunk 2]
```diff
@@ -16,10 +16,19 @@ fi
...
+
+    # Pick the release tag you prefer,
+    # more detailed tags/versions can be found at https://android.googlesource.com/platform/tools/metalava/+refs
+    git checkout android-12.1.0_r5
```
Comment (RingerJK) [inline scripts/update_metalava.sh:27]: that what I was looking for 👍

**Review by RingerJK — APPROVED** [review]

**Comment by Guardiola31337** [comment]
Going ahead and merging here 🚀 

cc @LukasPaczos @RingerJK @cafesilencio

File: gradle/metalava-dependencies.gradle [hunk 3]
```diff
@@ -6,18 +6,19 @@ dependencies {
...
-    metalava "com.android.tools.lint:lint:27.2.0-alpha11"
-    metalava "com.android.tools:common:27.2.0-alpha11"
-    metalava "com.android.tools:sdk-common:27.2.0-alpha11"
-    metalava "com.android.tools:sdklib:27.2.0-alpha11"
-    metalava "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.4.20"
-    metalava "org.jetbrains.kotlin:kotlin-reflect:1.4.20"
+    metalava "com.android.tools.external.org-jetbrains:uast:30.3.0-alpha08"
+    metalava "com.android.tools.external.com-intellij:kotlin-compiler:30.3.0-alpha08"
+    metalava "com.android.tools.external.com-intellij:intellij-core:30.3.0-alpha08"
+    metalava "com.android.tools.lint:lint-api:30.3.0-alpha08"
```
Comment (kmadsen) [inline gradle/metalava-dependencies.gradle:12]: Why use alpha versions? 

https://mvnrepository.com/artifact/com.android.tools.lint/lint-api?repo=google
should we use [30.2.1](https://mvnrepository.com/artifact/com.android.tools.lint/lint-api/30.2.1) instead?

Comment (Guardiola31337) [reply gradle/metalava-dependencies.gradle:12]: This is auto-generated in order to update Metalava https://github.com/mapbox/mapbox-navigation-android/blob/main/docs/metalava.md#to-update-metalava and by the way Metalava it's not even stable 😬

Comment (kmadsen) [reply gradle/metalava-dependencies.gradle:12]: oh right, update is automatic through the build https://android.googlesource.com/platform/tools/metalava/ 

I'm not seeing anywhere there is a place to pull the stable channel 🤔 . but of course the main line is an alpha
https://android.googlesource.com/platform/tools/metalava/+/HEAD/DOWNLOADING.md

Comment (Guardiola31337) [reply gradle/metalava-dependencies.gradle:12]: Yeah, wondering if we should keep using a Google / Android "internal" tool i.e. docs are poor and not well maintained, changes in the repo are really hard to follow, as it's still in alpha they keep breaking SemVer which makes updating it very difficult sometimes...

We may want to consider something else 🤔

Also noting that the Maps team already integrates [Binary compatibility validator](https://github.com/Kotlin/binary-compatibility-validator) https://github.com/mapbox/mapbox-maps-android/pull/1120.
